merchant/src/catalog/backend.rs

```diff
@@ -158,6 +219,98 @@ impl CatalogService for CatalogSQLService {
+        let mut objects_dependency_document: HashMap<String, &CatalogObject<String>> =
+            HashMap::new();
+
+        // we count all references to the dependency
+
+        for (index, item) in catalog.iter().enumerate() {
+            let key_id = match &item.id {
+                Some(id) => make_id_by_alias(id),
+                None => make_id_by_index(index.try_into().unwrap()),
+            };
+
+            objects_dependency.entry(key_id.clone()).or_insert(0);
+            objects_dependency_document
+                .entry(key_id)
+                .or_insert(&item.catalog_object);
+
+            match &item.catalog_object {
+                CatalogObject::Variation(entry) => {
+                    let key = make_id_by_alias(&entry.item_id);
+                    objects_dependency.entry(key).and_modify(|e| *e += 1);
+                }
+                CatalogObject::Modification(entry) => {
+                    let key = make_id_by_alias(&entry.item_id);
+                    objects_dependency.entry(key).and_modify(|e| *e += 1);
+                }
+                _ => {}
+            }
+        }
+
+        let mut items_sorted_to_insert: Vec<(&String, &u32)> = objects_dependency.iter().collect();
+        items_sorted_to_insert.sort_by(|a, b| b.1.cmp(a.1));
+        // we start creating the dependencies from the less dependant
+        for (alias_id, _) in items_sorted_to_insert.clone() {
+            let item = *objects_dependency_document.get(alias_id).unwrap();
+            match item {
+                CatalogObject::Item(item) => {
+                    let catalog_object: CatalogObject<self::Id> =
+                        CatalogObject::<u32>::Item(item.clone());
+                    let document = self.create(account.clone(), &catalog_object).await?;
+                    objects_created.entry(alias_id.clone()).or_insert(document);
+                }
+                CatalogObject::Variation(variation) => {
+                    if let Some(catalog_object) = objects_created.get(&variation.item_id.clone()) {
+                        let catalog_object =
+                            <Self as BulkDocumentConverter>::to_simple_catalog_object(
+                                catalog_object.id,
+                                &CatalogObject::Variation(variation.clone()),
+                            )
+                            .unwrap();
+                        let document = self.create(account.clone(), &catalog_object).await?;
+                        objects_created.entry(alias_id.clone()).or_insert(document);
+                    } else {
+                        return Err(CatalogError::BulkReferenceNotExist(
+                            variation.item_id.clone(),
+                        ));
+                    }
+                }
+                CatalogObject::Modification(modification) => {
+                    if let Some(catalog_object  ) = objects_created.get(&modification.item_id.clone())
```

Comment:
```suggestion
                    if let Some(catalog_object) = objects_created.get(&modification.item_id.clone())
```
Are we missing a `cargo fmt`?